### PR TITLE
Add Dokka documentation to artifacts

### PR DESCRIPTION
**Pull request by tripolkaandrey**
This PR adds Dokka-generated documentation as a publish artifact. 

It is done by using functionality introduced in [this PR in config](https://github.com/SpineEventEngine/config/pull/352).

Also, while updating the config, the dead code from buildSrc, like the `groovy` directory, was removed.

## Important details
Breaking changes were introduced in the `2.11.0` version of `error_prone_core` leading to a compilation error. The `2.10.0` version was forced for all [other](https://mvnrepository.com/artifact/com.google.errorprone/error_prone_core/2.10.0) Error Prone parts to resolve the issue.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -110,20 +115,35 @@ all modules and does not describe the project structure per-subproject.
     <artifactId>org.jacoco.ant</artifactId>
     <version>0.8.7</version>
   </dependency>
+  <dependency>
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-base</artifactId>
+    <version>1.6.20</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:123]: @sonatype-lift ignore

File: pom.xml [hunk 2]
```diff
@@ -62,7 +62,7 @@ all modules and does not describe the project structure per-subproject.
   <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:62]: @sonatype-lift ignore

File: pom.xml [hunk 3]
```diff
@@ -72,7 +72,7 @@ all modules and does not describe the project structure per-subproject.
...
     <groupId>com.google.protobuf</groupId>
     <artifactId>protoc</artifactId>
-    <version>3.19.4</version>
+    <version>3.20.0</version>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:77]: @sonatype-lift ignore

File: pom.xml [hunk 4]
```diff
@@ -110,20 +115,35 @@ all modules and does not describe the project structure per-subproject.
     <artifactId>org.jacoco.ant</artifactId>
     <version>0.8.7</version>
   </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:118]: @sonatype-lift ignore

File: pom.xml [hunk 5]
```diff
@@ -110,20 +115,35 @@ all modules and does not describe the project structure per-subproject.
...
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-base</artifactId>
+    <version>1.6.20</version>
+  </dependency>
+  <dependency>
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-core</artifactId>
+    <version>1.6.20</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:128]: @sonatype-lift ignore

**Comment by tripolkaandrey** [comment]
Vulnerabilities regarding Dokka dependencies can be fixed only by adding dependency constraints in the [config](https://github.com/SpineEventEngine/config) repository because they originated from transitive dependencies of the precompiled script [dokka-for-java](https://github.com/SpineEventEngine/config/blob/master/buildSrc/src/main/kotlin/dokka-for-java.gradle.kts).

The vulnerability of error_prone_core is not relevant as this dependency is forced to [2.10.0](https://mvnrepository.com/artifact/com.google.errorprone/error_prone_core/2.10.0) which uses [30.1-jre](https://mvnrepository.com/artifact/com.google.guava/guava/30.1-jre) with no mentioned vulnerabilities.

**Review by armiol — APPROVED** [review]
LGTM.

**Pull request merged**